src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -231,12 +231,33 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
 
   antlrcpp::Any visitGroupClause(
       SparqlAutomaticParser::GroupClauseContext* ctx) override {
-    return visitChildren(ctx);
+    vector<GroupKey> groupConditions;
+    for (auto* groupCondition : ctx->groupCondition()) {
+      groupConditions.push_back(
+          visitGroupCondition(groupCondition).as<GroupKey>());
+    }
+    return groupConditions;
   }
 
   antlrcpp::Any visitGroupCondition(
       SparqlAutomaticParser::GroupConditionContext* ctx) override {
-    return visitChildren(ctx);
+    if (ctx->var() && !ctx->expression()) {
+      return GroupKey{VariableGroupKey(ctx->var()->getText())};
+    } else if (ctx->builtInCall() || ctx->functionCall()) {
+      // builtInCall and functionCall are both also an Expression
+      auto expr = sparqlExpression::SparqlExpressionPimpl{
```

Comment:
It looks like accesses the `nullptr` `builtInCall` if there actually was a functionCall... you have to properly dispatch here.

Reply:
You're right.